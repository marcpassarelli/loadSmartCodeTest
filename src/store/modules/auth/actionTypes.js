const actionTypes = {
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_FAILED: 'LOGIN_FAILED',
  LOGOUT: 'LOGOUT',
};

export default actionTypes;
