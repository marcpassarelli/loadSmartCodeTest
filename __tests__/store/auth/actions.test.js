import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock-jest';
import expect from 'expect'; // You can use any testing library
import actionTypes from '../../../src/store/modules/auth/actionTypes';
import {
  getNewAccessToken,
  loginFailed,
  loginSuccess,
} from '../../../src/store/modules/auth/actions';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe('async actions', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  it('creates LOGIN_SUCCESS when authentication has been done', () => {
    const store = mockStore({});

    return store.dispatch(getNewAccessToken()).then(() => {
      expect(store.getActions()[0]).toEqual(loginSuccess());
      expect(store.getActions()[0]).toMatchSnapshot();
    });
  });

  it('creates LOGIN_FAILED when authentication has been wrong', () => {
    const store = mockStore({});
    const error = {
      error: undefined,
      type: actionTypes.LOGIN_FAILED,
    };

    return store
      .dispatch(getNewAccessToken())
      .then(() => Promise.reject(error))
      .catch((err) => {
        expect(err).toEqual(loginFailed());
        expect(err).toMatchSnapshot();
      });
  });
});
