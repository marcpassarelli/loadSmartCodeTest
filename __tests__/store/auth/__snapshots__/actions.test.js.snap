// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates LOGIN_FAILED when authentication has been wrong 1`] = `
Object {
  "error": undefined,
  "type": "LOGIN_FAILED",
}
`;

exports[`async actions creates LOGIN_SUCCESS when authentication has been done 1`] = `
Object {
  "authInfo": undefined,
  "type": "LOGIN_SUCCESS",
}
`;
